test/tests/api/v1_1/nodes_tests.py

```diff
@@ -92,21 +94,28 @@ def __post_workflow(self, id, graph_name, data):
             sleep(1)
             timeout -= 1
         return timeout
-
-    def check_compute_count(self):
+    
+    def get_undiscovered_compute_count(self):
         Nodes().nodes_get()
         nodes = self.__get_data()
-        count = 0
+        discovered_count = 0
         for n in nodes:
             type = n.get('type')
             if type == 'compute':
-                count += 1
-        return count
+                id = n.get('id')
+                status = self.__get_workflow_status(id)
+                if status != 'pending' and status != 'running':
```

Comment:
IMHO, I think the logic is a bit unnecessary.  Something like https://github.com/jlongever/RackHD/commit/acda755022920ba4ed90313a3a5cd176b6e70357 but include node catalog checks if the test is using a fixed count?